osu.Game/Beatmaps/BeatmapManager.cs

```diff
@@ -414,6 +415,9 @@ private bool fetchAndPopulateOnlineIDs(BeatmapInfo beatmap, IEnumerable<BeatmapI
 
                 Logger.Log($"Successfully mapped to {res.OnlineBeatmapSetID} / {res.OnlineBeatmapID}.", LoggingTarget.Database);
 
+                beatmap.Status = res.Status;
```

Comment:
this should be blow the `if`

Reply:
This is done because the status is not part of the beatmap file, whereas OnlineBeatmapID/OnlineBeatmapSetID are.